### PR TITLE
[MVS-446] Create submit vaccination endpoint

### PR DESCRIPTION
**Pull request by axn9t9**
## :floppy_disk: [Code Check]
_Make sure you've checked/done the following before making this PR!_
* Does your code build/run?
* Does your design approach make sense?
* Does your work achieve the intention of the card?
* Does your code have any unintentional side effects -- have you tested a full workflow & checked the logic branches you touched?
* Did you make a unit test for your code (as appropriate)?
* Is your branch relatively up to date with the Development branch (within ~2 commits)
* Have you moved the card to "Needs QA" in Jira

## :flipper: [JIRA Sprint Card]

https://mass-immunization-system.atlassian.net/browse/MVS-446

## :vertical_traffic_light: [Testing/QA Notes]

- Run docker containers
- Create preliminary FHIR resources (referenced FHIR resources must exist)
- Send a POST request to /submit-vaccination with the request data found in models/Immunization and models/Observation

> {
>   "Immunization": {},
>   "Observation": {}
> }

- Ensure the response data contains all of the provided information.

### CODE REVIEW

File: broker/server.js [hunk 1]
```diff
@@ -104,6 +114,39 @@ app.post("/Encounter", (req, res) => {
...
+
+  // post resource
+  postImmunization(imm)
+    .then((imm_res) => {
+      response.Immunization = imm_res;
+
+      obs.partOf = `Immunization/${imm_res.id}`;
+      postObservation(obs)
+        .then((obs_res) => {
+          response.Observation = obs_res;
```
Comment (rrucker3) [inline broker/server.js:133]: Can you also get the reaction field of the Immunization to point to the newly created Observation? You can use a https://www.hl7.org/fhir/http.html#patch request to update a specific field in an existing resource.

**Review by rrucker3 — CHANGES_REQUESTED** [review]
I wasn't able to get a successful test. First, I copied the objects from Immunization.js and Observation.js, but I got this error: "Can not create resource with ID[3303], no resource with this ID exists and clients may only assign IDs which contain at least one non-numeric character"

Then I tried removing the reaction element from the Immunization object, but that failed too, with this error: imm.reaction is not iterable.

I put reaction back in as [""], but then I got this error: "Can not create resource with ID[715], no resource with this ID exists and clients may only assign IDs which contain at least one non-numeric character"

Bottom line, I think the testing instructions need to be more detailed. I can't just take the models from the models files and have them work.

**Comment by TomMakes** [comment]
I'm not seeing any places where you ensure the response data contains all the necessary info.

**Review by rrucker3 — APPROVED** [review]
